pkg/storaged/fsys-panel.jsx

```diff
@@ -0,0 +1,121 @@
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { StorageUsageBar } from "./storage-controls.jsx";
+import { decode_filename, block_name, fmt_size, format_fsys_usage, go_to_block } from "./utils.js";
+
+const _ = cockpit.gettext;
+
+export class FilesystemsPanel extends React.Component {
+    constructor () {
+        super();
+        this.on_fsys_samples = () => { this.setState({}); }
+    }
+
+    componentDidMount() {
+        this.props.client.fsys_sizes.addEventListener("changed", this.on_fsys_samples);
+    }
+
+    componentWillUnmount() {
+        this.props.client.fsys_sizes.removeEventListener("changed", this.on_fsys_samples);
+    }
+
+    render() {
+        var client = this.props.client;
+
+        function is_mount(path) {
+            var block = client.blocks[path];
+            var fsys = client.blocks_fsys[path];
+            return fsys && block.IdUsage == "filesystem" && block.IdType != "mpath_member" && !block.HintIgnore;
+        }
+
+        function cmp_mount(path_a, path_b) {
+            var name_a = client.blocks[path_a].IdLabel || block_name(client.blocks[path_a]);
+            var name_b = client.blocks[path_b].IdLabel || block_name(client.blocks[path_b]);
+            return name_a.localeCompare(name_b);
+        }
+
+        function make_mount(path) {
+            var block = client.blocks[path];
+            var fsys = client.blocks_fsys[path];
+            var mount_points = fsys.MountPoints.map(decode_filename);
+            var fsys_size;
+            for (var i = 0; i < mount_points.length && !fsys_size; i++)
+                fsys_size = client.fsys_sizes.data[mount_points[i]];
+
+            function go(event) {
+                if (!event || event.button !== 0)
+                    return;
+                go_to_block(client, path);
+            }
+
+            return (
+                <tr onClick={go}>
+                    <td>{ block.IdLabel || block_name(block) }</td>
+                    <td>
+                        { fsys.MountPoints.length > 0 ?
+                          fsys.MountPoints.map((mp) => <div>{decode_filename(mp)}</div>)
+                              : "-"
+                        }
+                    </td>
+                    <td>
+                        { fsys.MountPoints.length > 0 ?
+                          <StorageUsageBar total={fsys_size? fsys_size[1] : 1}
+                                           used={fsys_size? fsys_size[0] : 0}
```

Comment:
The fallback values are a bit weird here. If the size isn't known, why is "1" the fallback? Also, "0 used" might theoretically be a valid value. Should these perhaps use `null` or `NaN` for unknown values? This pattern also occurs further down in other StorageUsageBar instantiations.

Reply:
The fallback is "0 bytes out of 1 byte used", but I agree that StorageUsageBar should do the fallback itself.

Reply:
Done.